ranges/ptr_arith.nim

```diff
@@ -8,3 +8,7 @@ proc distance*(a, b: pointer): int {.inline.} =
 
 proc shift*[T](p: ptr T, delta: int): ptr T {.inline.} =
   cast[ptr T](shift(cast[pointer](p), delta * sizeof(T)))
+
+template makeOpenArray*(p: pointer, T: type, len: int): auto =
```

Comment:
It's handier to just use `ptr T` instead of `pointer`, so that `makeOpenArray` requires less arguments. Also I'd rename it to `toOpenArray` for consistency with std.

Reply:
OK, I'll add the `ptr T` version too. These procs are a bit more unsafe than `toOpenArray`, so that's why I didn't want to use the more innocent name.

The bigger issue right now is that Nim 0.19 doesn't have the required magics. Did you had an alternative definition that works in 0.19? I pushed an ugly fallback that uses a sequence.

Reply:
> Did you had an alternative definition that works in 0.19?

Sure, see `template toOpenArray` in `typedranges.nim`.

Reply:
Yeah, I tried this, but it didn't work for some reason:

```nim
template makeOpenArray*[T](p: ptr T, len: int): auto =
  toOpenArray(cast[ptr array[0, T]](p)[], 0, len - 1)
```

Gives me `ttypedranges.nim(251, 15) Error: internal error: genMagicExpr: mSlice` at the call-site. Quite strange, because I don't see any difference.

Reply:
Oh it seems like `len` problem. Otherwise it works for me. Can be passed to functions, etc.

Reply:
Final code pushed, kudos for finding the issue :)